### PR TITLE
[ALLUXIO-2110] Improve BlockMaster unit tests

### PR DESCRIPTION
**Pull request by aaudiber**
https://alluxio.atlassian.net/browse/ALLUXIO-2110

### CODE REVIEW

File: core/server/src/main/java/alluxio/master/AbstractMaster.java [hunk 1]
```diff
@@ -65,13 +58,18 @@
   /** The {@link AsyncJournalWriter} for async journal writes. */
   private AsyncJournalWriter mAsyncJournalWriter = null;
 
+  /** The clock to use for determining the time. */
+  protected final Clock mClock;
+
   /**
    * @param journal the journal to use for tracking master operations
    * @param numThreads the number of threads to use in the Master's {@link ExecutorService}
```
Comment (apc999) [inline core/server/src/main/java/alluxio/master/AbstractMaster.java]: out of date

Comment (aaudiber) [inline core/server/src/main/java/alluxio/master/AbstractMaster.java]: Fixed

**Comment by aaudiber** [comment]
@apc999 All comments addressed, PTAL

File: core/server/src/main/java/alluxio/master/AbstractMaster.java [hunk 2]
```diff
@@ -65,13 +58,18 @@
...
+  protected final Clock mClock;
+
   /**
    * @param journal the journal to use for tracking master operations
-   * @param numThreads the number of threads to use in the Master's {@link ExecutorService}
+   * @param clock the Clock to use for determining the time
+   * @param executorService the executor service to use for running maintenance threads
    */
-  protected AbstractMaster(Journal journal, int numThreads) {
+  protected AbstractMaster(Journal journal, Clock clock, ExecutorService executorService) {
```
Comment (gpang) [inline core/server/src/main/java/alluxio/master/AbstractMaster.java]: can `executorService` be null?

File: core/server/src/main/java/alluxio/master/AbstractMaster.java [hunk 3]
```diff
@@ -178,19 +171,16 @@ public void stop() throws IOException {
...
-      String awaitFailureMessage =
-          "waiting for {} executor service to shut down. Daemons may still be running";
-      try {
-        if (!mExecutorService.awaitTermination(SHUTDOWN_TIMEOUT_MS, TimeUnit.MILLISECONDS)) {
-          LOG.warn("Timed out " + awaitFailureMessage, this.getClass().getSimpleName());
-        }
-      } catch (InterruptedException e) {
-        LOG.warn("Interrupted while " + awaitFailureMessage, this.getClass().getSimpleName());
+    // Shut down the executor service, interrupting any running threads.
+    mExecutorService.shutdownNow();
```
Comment (gpang) [inline core/server/src/main/java/alluxio/master/AbstractMaster.java]: is this idempotent?

**Comment by gpang** [comment]
@aaudiber Thanks! I left a few small comments.

File: core/server/src/main/java/alluxio/master/AbstractMaster.java (hunk 2 above)
Comment (aaudiber) [inline core/server/src/main/java/alluxio/master/AbstractMaster.java]: It shouldn't be, I've added checkNotNull

File: core/server/src/main/java/alluxio/master/AbstractMaster.java (hunk 3 above)
Comment (aaudiber) [inline core/server/src/main/java/alluxio/master/AbstractMaster.java]: yes

File: core/common/src/test/java/alluxio/clock/TestClock.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  /**
+   * Constructs a TestClock set to the specified time.
+   *
+   * @param time the time to set the clock to
+   */
+  public TestClock(long time) {
+    mTime = time;
+  }
+
+  public synchronized void setTimeMs(long time) {
```
Comment (apc999) [inline core/common/src/test/java/alluxio/clock/TestClock.java]: javadoc for this setter?

Comment (aaudiber) [inline core/common/src/test/java/alluxio/clock/TestClock.java]: Updated, thanks

File: core/common/src/main/java/alluxio/clock/SystemClock.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.clock;
+
+/**
+ * The Java System clock.
+ */
+public class SystemClock implements Clock {
```
Comment (apc999) [inline core/common/src/main/java/alluxio/clock/SystemClock.java]: a separate question: do we still stick to apply the `final` keyword when possible?

File: core/common/src/main/java/alluxio/clock/SystemClock.java [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.clock;
+
+/**
+ * The Java System clock.
```
Comment (apc999) [inline core/common/src/main/java/alluxio/clock/SystemClock.java]: might be better to say a little bit more when are we supposed to use this class

File: core/common/src/main/java/alluxio/clock/SystemClock.java (hunk 5 above)
Comment (aaudiber) [inline core/common/src/main/java/alluxio/clock/SystemClock.java]: yes, updated

File: core/common/src/main/java/alluxio/clock/SystemClock.java (hunk 6 above)
Comment (aaudiber) [inline core/common/src/main/java/alluxio/clock/SystemClock.java]: done

**Comment by apc999** [comment]
LGTM.

thanks for improving the tests, the code is more healthy now

**Comment by aaudiber** [comment]
@gpang Comments addressed, PTAL

**Comment by gpang** [comment]
LGTM

**Pull request merged**